### PR TITLE
[WIP] misc. backwards compatibility tests

### PR DESCRIPTION
**Pull request by Sjors**
Builds on top of #12134.

Tests:
- [x] v0.16 generated wallets don't work on older versions
- [x] the five SegWit wallet scenario's (including bech32 address)  

Test framework changes (or: things I'm doing wrong):
- [ ] method to copy wallet
- [ ] method to push and pop wallet backups
- [ ] remember network topology between start and restart

Suggestions for additional tests welcome. I'll make individual PR's later.

### CODE REVIEW

File: contrib/devtools/previous_release.sh [hunk 1]
```diff
@@ -20,17 +22,23 @@ while getopts ":hfrd" opt; do
       echo "  -f   Configure for functional tests"
       echo "  -r   Remove existing directory"
       echo "  -d   Use depends"
+      echo "  -b   configure --with-incompatible-bdb"
```
Comment (Sjors) [inline contrib/devtools/previous_release.sh]: I'm not using this. Will remove unless someone wants to keep it.

File: test/functional/backwards_compatibility.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        self.nodes[1].getnewaddress(account="p2sh-disappear", address_type='bech32') # wrong type!
+        self.nodes[1].getnewaddress(account="p2sh-disappear", address_type='p2sh-segwit') # next key...
+        assert_not_equal(self.nodes[1].getaddressesbyaccount("p2sh-disappear"), [address7])
+        assert_equal(self.nodes[1].getbalance("p2sh-disappear"), 0)
+
+        # Balance won't immedidately update:
+        self.nodes[1].rescanblockchain()
+        assert_equal(self.nodes[1].getbalance("p2sh-post-backup"), 1)
+        assert_equal(self.nodes[1].getbalance("bech32-post-backup"), 1) 
+        assert_equal(self.nodes[1].getbalance("p2sh-disappear"), 0)
```
Comment (Sjors) [inline test/functional/backwards_compatibility.py]: @sipa I'm not sure how to interpret scenario 1. It seems that, for keys generated after a backup was made, funds are irretrievable if the user doesn't create the correct address type for each new key. Is this a bug?

**Comment by Sjors** [comment]
I need to rewrite these tests so they don't use the account feature. See also [IRC log](https://botbot.me/freenode/bitcoin-core-dev/2018-01-12/?msg=95679098&page=3).

**Comment by fanquake** [comment]
Given that this has been work in progress for a few months, and "I need to rewrite these tests so they don't use the account feature", I'm closing this for now. Can be re-opened after it's been re-written and rebased. Review can be focussed on #12134 as well.

**Pull request closed without merging**